src/sql/orderbook/batch_rewards.sql

```diff
@@ -0,0 +1,40 @@
+with hashed_observations as (SELECT block_number,
+                                    tx_hash,
+                                    effective_gas_price * gas_used as execution_cost,
+                                    surplus,
+                                    fee,
+                                    auction_id
+                             FROM settlement_observations so
+                                    LEFT OUTER JOIN LATERAL (
+                               SELECT tx_hash, solver, tx_nonce, tx_from
+                               FROM settlements s
+                               WHERE s.block_number = so.block_number
+                                 AND s.log_index > so.log_index
```

Comment:
Why is `>` here?

Reply:
Settlement event is logged after all trades have gone through at the end of the transaction (as the last event). It is technically possible that two batches are settled in the same transaction so we need to match the correct trades with the correct settlements.

Reply:
Oh, you expect `log_index` in `settlement_observations` to be index of the trade event, not the index of settlement index?

Reply:
No.. I expect it to be the log index of the settlement event of course. I mean this is a pretty common join being made... but maybe I have misinterpreted.

Maybe we should cross reference with this.

https://github.com/cowprotocol/services/blob/ac5ac7365a03ffb44a44d0c840b16396f66c9fb6/crates/database/src/trades.rs#L28-L48

In the case of this query -- we aren't actually joining on trade events so perhaps the fancy join isn't necessary. Would be a lot easier to validate this if there were already populated data...

Reply:
Actually this may not be necessary. What matters to me here is that we need a SINGLE `tx_hash` for a batch but it is possible that there are two settlements in the same transaction. I really don't know how to deal with this just yet, but I do think that maybe this query needs to be tested and fixed.

Reply:
Maybe we don't need to do the lateral join at all. Problem is this will all break if there are two settlement in the same transaction.

Reply:
OK I see the issue here. We should not be using inequality, but rather equality on the log index. Sorry for the confusion (this query was partially copied from the join on trades above).

Reply:
It is very good that you have pointed this out. I would like to get some more eyes on this. How will the services handle the scenario when there are two settlements in the same tx?

Reply:
Cross reference you mentioned is indeed related to trade events that have their own `log_index` which is different (lower) than settlement `log_event` so I _think_  this does not apply in our case, where we simply use the same event in both tables.

> How will the services handle the scenario when there are two settlements in the same tx?

Have no idea, will think about it.

Reply:
Ok, so I have dramatically simplified this query for the time being and would propose we assume this is good for now and circle back to fix it as necessary once the data actually becomes available.